src/fetch.ts

```diff
@@ -144,3 +144,24 @@ export const retriableFetch: (
     }, reject);
   });
 };
+
+/**
+ * Makes fetch able to process response with a given processor function
+ *
+ * @param processor: a function that applies to original response
+ * @returns: a new fetch with processor capabilities
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const fetchWithProcessor = (processor: (a: any) => any) => (
```

Comment:
Can we add a type? The type is useful to write the right logic inside the processor. 
```suggestion
export const fetchWithProcessor = <T>(processor: (a: unknown) => T) => (
```
A general question. How this processor will be used? We often pass the fetch to the generated Client that serve many endpoint with different responses and payloads.

Reply:
This processor can be used as a default behaviour when you have to "clean" the response catched by the original fetch operation. Because of our openapi-codegen strict typings, wrong 3rd parties API specs can cause decoding problems while dealing with them. So this processor allows to reconcile expected response shape :)